### PR TITLE
Add operation tests about buffer zero initialization - Part III

### PR DESCRIPTION
**Pull request by Jiawei-Shao**
<hr>

**Author checklist for test code/plans:**

- [ ] All outstanding work is tracked with "TODO" in a test/file description or `.unimplemented()` on a test.
- [ ] New helpers, if any, are documented using `/** doc comments */` and can be found via `helper_index.txt`.
- [ ] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)

**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [ ] The test path is reasonable, the [description](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) "describes the test, succinctly, but in enough detail that a reader can read only the test plans in a file or directory and evaluate the completeness of the test coverage."
- [ ] Tests appear to cover this area completely, except for outstanding TODOs. Validation tests use control cases.
    (This is critical for coverage. Assume anything without a TODO will be forgotten about forever.)
- [ ] Existing (or new) test helpers are used where they would reduce complexity.
- [ ] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

File: src/webgpu/api/operation/resource_init/buffer.spec.ts [hunk 1]
```diff
@@ -309,3 +308,93 @@ remaining part of it will be initialized to 0.`
...
+    for (let layer = 0; layer < arrayLayerCount; ++layer) {
+      const renderPass = encoder.beginRenderPass({
+        colorAttachments: [
+          {
+            view: srcTexture.createView({
+              baseArrayLayer: layer,
+              arrayLayerCount: 1,
+              baseMipLevel: copyMipLevel,
+            }),
+            loadValue: { r: 0.2 * (layer + 1), g: 0, b: 0, a: 0 },
```
Comment (Kangz) [inline src/webgpu/api/operation/resource_init/buffer.spec.ts:372]: Won't this clamp to 1 as soon as the second layer? To have less float precision issues maybe we could use an integer texture format.

**Review by Kangz — APPROVED** [review]

File: src/webgpu/api/operation/resource_init/buffer.spec.ts [hunk 2]
```diff
@@ -309,3 +308,93 @@ remaining part of it will be initialized to 0.`
...
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.RENDER_ATTACHMENT,
+      size: textureSize,
+      mipLevelCount: copyMipLevel + 1,
+    });
+
+    const bytesPerRow = 256;
+    const layout = getTextureCopyLayout(
+      srcTextureFormat,
+      '2d',
+      [textureSize.width, textureSize.height, textureSize.depthOrArrayLayers],
```
[the remark this answers is not in the document]
Comment (Jiawei-Shao) [reply src/webgpu/api/operation/resource_init/buffer.spec.ts:344]: Oh I can't do that because of the following error:

Argument of type '{ width: number; height: number; depthOrArrayLayers: number; }' is not assignable to parameter of type 'readonly [number, number, number]'

[the remark this answers is not in the document]
Comment (Jiawei-Shao) [reply src/webgpu/api/operation/resource_init/buffer.spec.ts]: Yes you are right!

**Review by Kangz — APPROVED** [review]
Still LGTM, feel free to merge.
